src/lib.rs

```diff
@@ -0,0 +1,50 @@
+pub use error::Error;
+use linter::clippy::Clippy;
+pub use linter::Lint;
+use std::env;
+
+mod error;
+mod git;
+mod linter;
+mod project;
+mod scout;
+
+pub struct ScoutOptions {
```

Comment:
These options seem to be intended for specific implementations of the generic `Linter` and `Config` traits (clippy and cargo project), which makes me think `ScoutOptions` should not be part of the library types, but I guess this is debatable.

We could imagine `scout` to be a library that both applications (1) `cargo-scout` and (e.g.) (2) `cpp-scout` use.
1) Would implement `Lint` for `Clippy` and `Config` for `(Cargo)Project`
2) Would implement `Lint` for `ClangTidy` and `Config` for `CMakeProject`
`Clippy` could be provided by `scout`, or it could be just part of the user application. 

Maybe `run` could be generic on the `Config` and `Project` traits and have the specific implementations injected instead of being instantiated inside of the function? Let me know your thoughts!

Reply:
Hmm I like where this is going !
Options would have associated types so options would match their specific linter.

I think we're not there yet, so I ll leave it as is and we'll write a followup